File: solutions/python/resistor-color-duo/2/resistor_color_duo.py
COLOR_LIST = [
    "black",
    "brown",
    "red",
    "orange",
    "yellow",
    "green",
    "blue",
    "violet",
    "grey",
    "white",
]


def value(colors: list[str]) -> int:
    """Return the resistor value from the first two color bands."""
    return 10 * COLOR_LIST.index(colors[0]) + COLOR_LIST.index(colors[1])

File: solutions/typescript/resistor-color-duo/3/resistor-color-duo.ts
const COLOR_MAP = {
  black: 0,
  brown: 1,
  red: 2,
  orange: 3,
  yellow: 4,
  green: 5,
  blue: 6,
  violet: 7,
  grey: 8,
  white: 9,
} as const;

type Color = keyof typeof COLOR_MAP;

export function decodedValue(colors: Color[]): number {
  return 10 * COLOR_MAP[colors[0]] + COLOR_MAP[colors[1]];
}
